groups/script.js

```diff
@@ -102,6 +111,35 @@ groupRoles?.addEventListener('click', function (event) {
   }
 });
 
+let paragraphElement, paragraphContent;
```

Comment:
n the declaration of the paragraphElement and paragraphContent variables. These variables are declared as let, which means that they are scoped to the current function. However, the debounce() function is being called from within the onclick event handler of the searchInput element. This means that the paragraphElement and paragraphContent variables will be created each time the searchInput element is clicked. This is not what you want, because you want these variables to be created once and then reused.

To fix the bug, you need to declare the paragraphElement and paragraphContent variables as const. This will make them global variables, and they will only be created once.

Reply:
Declared paragraphElement and paragraphContent variables as const.